pages/quiz/how-are-you-feeling.tsx

```diff
@@ -222,8 +232,20 @@ export const HowAreYouFeelingPage = (): JSX.Element => {
           homeButton={false}
           filterButton={false}
         />
+        <Navigation>
+          <LinkButton
+            textContent="back"
+            arrow="back"
+            onClick={() => router.back()}
+          />
+          <LinkButton
+            textContent="skip"
+            arrow="forward"
+            onClick={() => router.push('/quiz/what-are-your-interests')}
+          />
+        </Navigation>
```

Comment:
This can also be a shared component, I think. Perhaps:

```jsx
<QuizNavigation
  isFirst (or isLast)
  nextPage="/quiz/whatever"
  lastPage="/quiz/whatever"
>
```

n.b., don't use `router.back()` - if the user starts a new session halfway through the quiz, it won't work.

Reply:
These can also just be links as we know their destinations too.

Reply:
Yep, I was just following the pattern we had in the component I abstracted it from

Reply:
@jthistle There are quite a lot of places where we duplicate styles, I think too much abstraction can often make the code more difficult to read and debug in the future. You might save a few lines of code in a few files, but is it worth it for the cost of the extra dev time involved in not only refactoring, but also in potentially obfuscating meaning for those understanding the code in the future?

Reply:
Good point about using router.back() though, I'll make the changes

Reply:
@edpe yes, there's a balance, and different people prefer different things. I tend to stray nearer to the 'modularise everything' camp.

Why? Well,

- you say it 'obfuscates' code, but really, if you chose a good name for your component, I'd argue it's more readable. You get a broader overview of the structure of the code in fewer characters, meaning that when you want to find something in particular,  you don't have to read through every unnecessary detail.
- when something inevitably breaks / NPC request a change to styles for e.g. all buttons, you _save_ the dev time you lose by making it modular in the first place, since you don't have to go through ever instance of the component to make a fix. And then, you potentially miss some occurences, leading to a new ticket and yet more dev time spent on it.

Those are the two main points; there are other benefits (and yes, a few disadvantages - if you modularise things which are only used once or twice, you'll end up with more problems, since if the components are that infrequent they probably had special purposes and you'll need to unmodularise them in the future to implement some new functionality. I don't think the navigation here falls into that category.)

And, just because we already have code duplication doesn't mean it's necessarily a good thing. I remember having to change something about button styling, requiring me to go through loads of files because we use buttons everywhere, and ended up making `<ButtonBase>` as a result.

Anyway, up to you whether you want to make this thing a component or not :upside_down_face:

Reply:
In the grand scheme of things, none of this matters anyway so I'll approve it whatever you do :)

Reply:
Working in components is probably the best thing about React. How much we do this is a preference for sure, but I think it works best with larger page elements when a project is quite early. The Navigation feels like the thing to abstract here - whatever it looks like and is made from internally, we know we want the same thing on 5 pages.

I'm less keen to abstract styling as this is still changing quite rapidly, and elements are often supposed to be unique/specific for now. A good place for enforcing style consitency is in the theme. There are defaults, and colour, font and breakpoint restrictions there but we could add more things like border-radii and padding options.